inst/fabric/xaringanExtra-fabric.js

```diff
@@ -0,0 +1,409 @@
+ */
+
+// Global Variables
+var mouseDown = false
+var fabrics = new Array()
+var fabricID
+var drawMode = false
+var eraseMode = false
+
+window.xaringanExtraFabric = function (opts) {
+  // User provided options
+  let defaultPenColor = opts['pen_color'][0]
+  chosenStrokeColor = defaultPenColor
+  let defaultPenSize = parseInt(opts['pen_size'][0])
+  let defaultEraserSize = parseInt(opts['eraser_size'][0])
+  let defaultEraserColor = opts['eraser_color'][0]
+  let tolerance = defaultEraserSize
+
+  document.addEventListener('DOMContentLoaded', function () {
+    // Get visible slide
+    const getVisibleSlide = function () {
+      const visible = document.querySelector('.remark-visible')
+      return visible
+    }
+
+    // Get visible slide dimensions
+    const getVisibleSlideSize = function () {
+      const visible = getVisibleSlide()
+      const slideSize = visible
+        .querySelector('.remark-slide-scaler')
+        .getBoundingClientRect()
+      return slideSize
+    }
+
+    // Get visible slide scaling ratio
+    const getVisibleSlideScaleRatio = function () {
+      const visible = getVisibleSlide()
+      const scaleRatio = visible.querySelector('.remark-slide-scaler').style
+        .transform
+      return scaleRatio
+    }
+
+    // Get visible slide Canvas
+    const getVisibleSlideCanvas = function () {
+      return fabrics[slideshow.getCurrentSlideIndex()]
+    }
+
+    // Get visible slide canvas container div
+    const getVisibleSlideOuterContainer = function () {
+      const visible = getVisibleSlide()
+      const outerDiv = visible.querySelector('.outer-canvas-container')
+      return outerDiv
+    }
+
+    // Get visible slide canvas container div
+    const getVisibleSlideCanvasContainers = function () {
+      const visible = getVisibleSlide()
+      const canvasDiv = visible.querySelectorAll('.drawing-canvas')
+      return canvasDiv
+    }
+
+    // Collect all paths on Canvas
+    const getCanvasPaths = function () {
+      return fabricID.getObjects('path')
+    }
+
+    const mousedown = function () {
+      mouseDown = true
+    }
+
+    // Canvas/canvas container constructor
+    const createCanvas = function (id) {
+      const slideSize = getVisibleSlideSize()
+
+      const canvasDiv = document.createElement('div')
+      canvasDiv.classList.add('outer-canvas-container')
+      canvasDiv.style.width = slideSize.width
+      canvasDiv.style.height = slideSize.height
+      canvasDiv.style.left = slideSize.left + 'px'
+      canvasDiv.style.top = slideSize.top + 'px'
+      ;['mousedown', 'touchstart'].forEach((gesture) => {
+        canvasDiv.addEventListener(gesture, mousedown)
+      })
+      ;['mouseup', 'touchend'].forEach((gesture) => {
+        canvasDiv.addEventListener(gesture, () => {
+          mouseDown = false
+          setTimeout(function () {
+            fabricID.forEachObject((obj) => {
+              obj.selectable = false
+              if (obj.stroke === defaultEraserColor) {
```

Comment:
This was definitely a first-pass hack solution. I added a stop condition in the `R`-side function to prohibit pen and eraser color from being the same, but this was a temporary guardrail. I was actually thinking about going back to making the eraser paths fully transparent (this would avoid pen color clashes), and just change the cursor to a circle to make it more obvious that erasing is "different" than drawing. Thoughts?

Reply:
Is it possible to simply remove the most recently-added path? Or to keep track of which paths are added during erasing and then remove them directly?

Reply:
Thoughts from @LauraRK: https://github.com/gadenbuie/xaringanExtra/pull/87#issuecomment-785283357

Reply:
> Is it possible to simply remove the most recently-added path? Or to keep track of which paths are added during erasing and then remove them directly?

Probably, I'll see what I can do.